### PR TITLE
Fix weaklist

### PR DESCRIPTION
**Pull request by vallsv**
- Fix sort
- Add reversed
- Add get/set/del slice
- Add str/repr

### CODE REVIEW

File: silx/utils/weakref.py [hunk 1]
```diff
@@ -305,9 +343,19 @@ def reverse(self):
...
+        Not very efficient.
+        """
+        sorted_list = list(self)
+        sorted_list.sort(key=key, reverse=reverse)
         self.__list = []
         self.extend(sorted_list)
+
+    def __str__(self):
+        unref_list = list(self)
+        return unref_list.__str__()
```
Comment (t20100) [inline silx/utils/weakref.py]: Maybe do`return 'WeakList(%s) % str(unref_list)`

File: silx/utils/weakref.py [hunk 2]
```diff
@@ -305,9 +343,19 @@ def reverse(self):
...
+
+    def __repr__(self):
+        unref_list = list(self)
+        return unref_list.__repr__()
```
Comment (t20100) [inline silx/utils/weakref.py]: Same, maybe do`return 'WeakList(%s) % repr(unref_list)`

**Comment by vallsv** [comment]
Yes, its a good idea

**Comment by vallsv** [comment]
Here it is

**Pull request merged**